pkg/transfers/pipeline/aggregate.go

```diff
@@ -271,6 +272,11 @@ func (xfagg *XferAggregator) await() chan error {
 		msg, err := xfagg.subscription.Receive(context.Background())
 		if err != nil {
 			xfagg.logger.LogErrorf("ERROR receiving message: %v", err)
+
+			if strings.Contains(err.Error(), "has been Shutdown") {
```

Comment:
I think we can use the context here to check for non-retryable errors:
```go
			if errors.Is(err, context.Canceled) {
				return
			}
```

Reply:
That doesn't seem to work. I'm not seeing the Subscription close cancel the Context.

Reply:
```go
func (xfagg *XferAggregator) await() chan error {
	out := make(chan error, 1)
	go func() {
		ctx, cancelFunc := context.WithCancel(context.Background())
		defer cancelFunc()

		msg, err := xfagg.subscription.Receive(ctx)
		if err != nil {
			xfagg.logger.LogErrorf("ERROR receiving message: %v", err)

			if errors.Is(err, context.Canceled) {
				// quit the infinite loop if it has been shutdown
				return
			}
		}
		out <- handleMessage(xfagg.merger, msg)
	}()
	return out
}
```

Reply:
Hmm, that's strange, maybe try:
```go
	select {
		case <-ctx.Done():
	           return
		}
```

Reply:
I can try, but what's wrong with the error comparison I had?

Reply:
The context needs to be pulled outside of the spawned goroutine. That'll allow the context to be captured, but I don't like how that's enclosed.

Reply:
The following will work
```go
		if gcerrors.Code(err) == gcerrors.FailedPrecondition {
			return
		}
```

Using `strings.Contains` works right now, but will break if the underlying error string ever changes.

Reply:
`FailedPrecondition` is the wrong error to check on though. I want to ensure it's a shutdown error. The library doesn't expose that string as an error to compare on.

Reply:
`pubsub` uses it as the shutdown code. 
https://github.com/google/go-cloud/blob/master/pubsub/pubsub.go#L669

But you're right, would be nice if they exposed the shutdown error..

I'd say continue on with `strings.Contains` or use a combination of both:
`if gcerrors.Code(err) == gcerrors.FailedPrecondition && strings.Contains(err.Error(), "shutdown")`

Reply:
Yea, that PreconditionFailed is a generic error in the library they use that implies authentication, shutdown, request overload, or other errors occur. I wish they would expose more granular error types. I created https://github.com/google/go-cloud/issues/2878 upstream for this.

Reply:
nice!

Reply:
fyi, they replied and `FailedPrecondition` would be a fine check here. I'm curious why it wasn't working yesterday then..

Reply:
Thanks for making the issue; the check `if gcerrors.Code(err) == gcerrors.FailedPrecondition` should work; I tested it yesterday.